ext/ffi/lib.rs

```diff
@@ -1117,6 +1245,25 @@ where
   Ok(ffi_args)
 }
 
+// SAFETY: Makes an FFI call
+unsafe fn ffi_call_rtype_struct(
+  size: usize,
+  cif: &libffi::middle::Cif,
+  fn_ptr: &libffi::middle::CodePtr,
+  call_args: Vec<Arg>,
+) -> NativeValue {
+  let mut result = vec![0u8; size].into_boxed_slice();
+  libffi::raw::ffi_call(
+    cif.as_raw_ptr(),
+    Some(*fn_ptr.as_safe_fun()),
+    result.as_mut_ptr() as *mut c_void,
+    call_args.as_ptr() as *mut *mut c_void,
+  );
+  NativeValue {
+    struct_value: Box::into_raw(result),
+  }
```

Comment:
May this accept a preallocated JS buffer instead?

On another note, does libffi care about alignment of the output buffer being less than that of the struct?